src/osd/scheduler/mClockScheduler.cc

```diff
@@ -425,18 +405,18 @@ void mClockScheduler::dump(ceph::Formatter &f) const
 void mClockScheduler::enqueue(OpSchedulerItem&& item)
 {
   auto id = get_scheduler_id(item);
-  auto op_type = item.get_op_type();
-  int cost = client_scaled_cost_infos[op_type];
-
-  // Re-calculate the scaled cost for the client if the item cost changed
-  if (maybe_update_client_cost_info(op_type, item.get_cost())) {
-    cost = client_scaled_cost_infos[op_type];
-  }
 
   // TODO: move this check into OpSchedulerItem, handle backwards compat
-  if (op_scheduler_class::immediate == item.get_scheduler_class()) {
+  if (op_scheduler_class::immediate == id.class_id) {
     immediate.push_front(std::move(item));
   } else {
+    auto op_type = item.get_op_type();
+    int cost = client_scaled_cost_infos[op_type];
+    // Re-calculate the scaled cost for the client if the item cost changed
+    if (maybe_update_client_cost_info(op_type, item.get_cost())) {
```

Comment:
have you tried always calling calc_scaled_cost? I wonder if part of the effect you're seeing with recovery is from using cached cost values that override a range of differences

Reply:
I tried calling the calc_scaled_cost always() with the latest changes (with pre-calculated osd_max_bandwidth) and pretty
much saw the same performance numbers. I do see the average client latency when using the "high_client_ops" profile to
be marginally better by a 10th of a millisecond in one of the runs. Therefore, it does look like using the cached cost values
doesn't help in the performance aspect. I removed the cached logic and always calculate cost now. I will re-post the
performance numbers with the latest change (which I suspect won't change much).